src/shared/services/auth/auth.js

```diff
@@ -70,6 +71,67 @@ function catchError() {
   }
 }
 
+function saveRefreshToken(refreshToken) {
+  storage.setItem(REFRESH_TOKEN_KEY, refreshToken);
+}
+
+/**
+ * Refresh access & refresh token
+ */
+async function refreshTokens() {
+  const { authEndpoint, realm, clientId } = configuration.keycloak;
+  const token = storage.getItem(REFRESH_TOKEN_KEY);
+
+  const searchParams = new URLSearchParams({
+    refresh_token: token,
+    grant_type: 'refresh_token',
+    client_id: clientId,
+    redirect_uri: AUTH_REDIRECT_URI,
+  });
+
+  const tokenUrl = `${authEndpoint}/realms/${realm}/protocol/openid-connect/token`;
+
+  try {
+    const response = await fetch(tokenUrl, {
+      body: searchParams.toString(),
+      method: 'POST',
+      headers: { 'Content-type': 'application/x-www-form-urlencoded' },
+    });
+
+    if (response.status === 200) {
+      const data = await response.json();
+      storage.setItem(ACCESS_TOKEN_KEY, data.access_token);
+      storage.setItem(REFRESH_TOKEN_KEY, data.refresh_token);
+    } else {
+      logout();
```

Comment:
This code means that if a user loses internet connection at the wrong time, he'll be logged out. There's a couple of other options: (1) ignore non-successful responses and wait for the next refresh interval to be triggered, (2) retry mechanism

Thoughts?